src/rely/QCheck_rely.re

```diff
@@ -0,0 +1,42 @@
+module T = QCheck.Test;
+module Rely = Rely;
+
+let seed_ =
+  lazy {
+    let s =
+      try (int_of_string @@ Sys.getenv("QCHECK_SEED")) {
+      | _ =>
+        Random.self_init();
+        Random.int(1000000000);
+      };
+    /* Printf.printf("qcheck random seed: %d\n%!", s); */
```

Comment:
Printing directly messed with the output from Rely so I commented it out. I imagine we will want to have the random seed somewhere and left some thoughts in the description of the PR.

Reply:
that makes sense, maybe a local file with the seed inside could be produced, or maybe some form of message (if rely provides such a mechanism)?

Reply:
I think writing to a temp file raises the same issue and I don't love just writing to the current directory (even only on failure), but could maybe be convinced.

What does a typical workflow look like with qcheck? If there is a failure is running repeatedly with a random seed still likely to produce the failure or is it best practice to hardcode the seed? If you hardcode a random seed when debugging are you reasonably likely to reproduce the failure (if so, how long would that take)?

Reply:
typically the runner displays the seed and provides a flag to set the seed (to one known to trigger some failure). In this case it should always reproduce the same failure if you don't use other sources of randomness. I don't like hardcoding the seed because using a fresh one every time covers more search space over time!